ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -259,8 +263,10 @@ private Long incrementalDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive
             Table table = hiveDb.getTable(dbName, tableName);
 
             // Dump external table locations if required.
-            if (shouldDumpExternalTableLocation() &&
-                    TableType.EXTERNAL_TABLE.equals(table.getTableType())) {
+            // Note: If repl policy is replaced, then need to dump external tables if table is getting replicated
+            // for the first time in current dump. So, need to check if table is included in old policy.
+            if ((shouldDumpExternalTableLocation() || !ReplUtils.tableIncludedInReplScope(work.oldReplScope, tableName))
```

Comment:
it should dump external table only if external table dump is enabled

Reply:
Good catch. Will fix it and add a test for this combination.